### PR TITLE
[steamapi] Export all types, add overloads and more

### PR DESCRIPTION
**Pull request by claabs**
- Make all declared types importable
- Add array overloads for `getUserBans` and `getUserSummary`
- Convert the declared structure classes to interfaces as they are just types and cannot be instantiated, and the structure classes are not directly exported by `steamapi`
- Add precise optionals for PlayerSummary (it's the only one I'm familiar with)
- Add missing getters for other structures

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: **Just some changes I felt it needed when working with the types myself**
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/steamapi/index.d.ts [hunk 1]
```diff
@@ -89,36 +90,37 @@ declare class SteamAPI {
...
+    getUserBadges(id: string): Promise<SteamAPI.PlayerBadges>;
 
     /**
      * Get users bans.
      * @param id User ID(s)
      * @returns Ban info
      */
-    getUserBans(id: string): Promise<PlayerBans | PlayerBans[]>;
+    getUserBans(id: string[]): Promise<SteamAPI.PlayerBans[]>;
+    getUserBans(id: string): Promise<SteamAPI.PlayerBans>;
```
Comment (joshuajeschek) [inline types/steamapi/index.d.ts:116]:
```suggestion
    /**
     * Get users bans.
     * @param id User ID
     * @returns Ban info
     */
    getUserBans(id: string): Promise<SteamAPI.PlayerBans>;
```

File: types/steamapi/index.d.ts [hunk 2]
```diff
@@ -139,200 +141,233 @@ declare class SteamAPI {
...
-    constructor(achievement: Record<string, unknown>);
-    api: string;
-    name: string;
-    description: string;
-    achievend: boolean;
-    unlockTime: number;
-
-    get unlockedAt(): Date;
+    getUserSummary(id: string[]): Promise<SteamAPI.PlayerSummary[]>;
+    getUserSummary(id: string): Promise<SteamAPI.PlayerSummary>;
```
Comment (joshuajeschek) [inline types/steamapi/index.d.ts:175]:
```suggestion
    /**
     * Get users summary.
     * @param id User ID
     * @returns Summary
     */
    getUserSummary(id: string): Promise<SteamAPI.PlayerSummary>;
```

**Review by joshuajeschek — COMMENTED** [review]
I think it'd be nice to add JSDoc to the overloads as well. The rest looks good to me :)

**Review by joshuajeschek — CHANGES_REQUESTED** [review]
Just for the bot, forgot to request changes

File: types/steamapi/index.d.ts (hunk 1 above)
Comment (claabs) [reply types/steamapi/index.d.ts:116]: JSDoc automatically propagates over multiple overloads, so it shouldn't be necessary.

VSCode:
image

File: types/steamapi/index.d.ts (hunk 2 above)
Comment (claabs) [reply types/steamapi/index.d.ts:175]: See: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57019#discussion_r744801503

File: types/steamapi/index.d.ts (hunk 1 above)
Comment (joshuajeschek) [reply types/steamapi/index.d.ts:116]: Didn't work for me 😔:

Comment (claabs) [reply types/steamapi/index.d.ts:116]: Must be IDE specific. I'll add the changes then. 😄

**Review by joshuajeschek — APPROVED** [review]
Perfect! Thanks for improving the types!

**Comment by claabs** [comment]
Ready to merge

**Pull request merged**